parlai/core/torch_agent.py

```diff
@@ -311,3 +323,26 @@ def shutdown(self):
         if path is not None:
             self.save(path + '.shutdown_state')
         super().shutdown()
+
+    def reset(self):
+        """Reset observation and episode_done."""
+        self.observation = None
+        self.episode_done = True
+
+    def observe(self, observation):
+        observation = copy.deepcopy(observation)
+        if not self.episode_done:
+            # if the last example wasn't the end of an episode, then we need to
+            # recall what was said in that example
+            prev_dialogue = self.observation['text']
+            observation['text'] = prev_dialogue + '\n' + observation['text']
+        self.observation = observation
+        self.episode_done = observation['episode_done']
+        return observation
```

Comment:
we should copy the more complex version of observe in parlai/agents/seq2seq.py which uses maintain_dialog_history instead of this simpler `'\n'`-concat.
(and by extension, remove self.episode_done as a field, e.g. in reset as well)
probably needs a little cleanup though
https://github.com/facebookresearch/ParlAI/blob/2951872be31f842cea861bcd52c34ed17e247f23/parlai/agents/seq2seq/seq2seq.py#L455-L475

Reply:
Alright, but it should it actually use the `maintain_dialogue_history` in this TorchAgent class, right?

Reply:
Going to put off this for another pull request, since this is just literally copying code from one class to another.